lib/BridgeManager.php

```diff
@@ -0,0 +1,624 @@
+		$pid = intval($pid);
+		if ($ret !== 0) {
+			$pid = 0;
+		}
+		return $pid;
+	}
+
+	/**
+	 * kill the mattermost processes (owned by web server unix user) that do not match with any room
+	 */
+	public function killZombieBridges() {
+		// get list of running matterbridge processes
+		$cmd = 'ps -ux | grep "commands/matterbridge" | grep -v grep | awk \'{print $2}\'';
+		exec($cmd, $output, $ret);
+		$runningPidList = [];
+		foreach ($output as $o) {
+			array_push($runningPidList, intval($o));
+		}
+		// get list of what should be running
+		$expectedPidList = [];
+		$this->manager->forAllRooms(function ($room) use (&$expectedPidList) {
+			$token = $room->getToken();
+			$bridge = $this->getBridgeOfRoom($token);
+			if ($bridge['enabled'] && $bridge['pid'] !== 0) {
+				array_push($expectedPidList, intval($bridge['pid']));
+			}
+		});
+		// kill what should not be running
+		foreach ($runningPidList as $runningPid) {
+			if (!in_array($runningPid, $expectedPidList)) {
+				$this->killPid($runningPid);
+			}
+		}
+	}
+
+	/**
+	 * Utility to kill a process
+	 *
+	 * @param int $pid the process ID to kill
+	 * @return bool if it was successfully killed
+	 */
+	private function killPid(int $pid): bool {
+		// kill
+		exec(sprintf('kill -9 %d', $pid), $output, $ret);
+		// check the process is gone
+		$isStillRunning = $this->isRunning($pid);
+		return (intval($ret) === 0 && !$isStillRunning);
+	}
+
+	/**
+	 * Check if a process is running
+	 *
+	 * @param int $pid the process ID
+	 * @return bool true if it's running
+	 */
+	private function isRunning(int $pid): bool {
+		try {
+			$result = shell_exec(sprintf('ps %d', $pid));
+			if (count(preg_split('/\n/', $result)) > 2) {
+				return true;
+			}
+		} catch (Exception $e) {
+		}
+		return false;
+	}
+
+	private function getBridgeFromDb(string $token): array {
+		$room = $this->manager->getRoomByToken($token);
+		$roomId = $room->getId();
+
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('json_values')
+			->from('talk_bridges', 'b')
+			->where(
+				$qb->expr()->eq('room_id', $qb->createNamedParameter($roomId, IQueryBuilder::PARAM_INT))
+			);
+		$req = $qb->execute();
+		$jsonValues = '{"enabled":false,"pid":0,"parts":[]}';
+		while ($row = $req->fetch()) {
+			$jsonValues = $row['json_values'];
+			break;
+		}
+		$req->closeCursor();
+		$qb = $qb->resetQueryParts();
+
+		return json_decode($jsonValues, true);
+	}
+
+	private function saveBridgeToDb(string $token, array $bridge) {
+		$room = $this->manager->getRoomByToken($token);
+		$roomId = $room->getId();
+		$jsonValues = json_encode($bridge);
+
+		$qb = $this->db->getQueryBuilder();
+
+		$exists = false;
+		$qb->select('json_values')
+			->from('talk_bridges', 'b')
+			->where(
+				$qb->expr()->eq('room_id', $qb->createNamedParameter($roomId, IQueryBuilder::PARAM_INT))
+			);
+		$req = $qb->execute();
+		while ($row = $req->fetch()) {
+			$exists = true;
```

Comment:
Instead of checking existance you could also insert and on `UniqueConstraintViolation` update.